src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure(int mode) {
+        Figure figure;
+        switch(mode) {
+            case 0: figure = new Circle(random.nextDouble());

```

Comment:
Instead of passing expressions directly into the constructor, create separate variables for the random values and pass those variables. This improves code readability.